#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_lin_velocity", double_t, 0, "Defines the maximum linear velocity", 1.0, 0.0, 5.0)
gen.add("max_ang_velocity", double_t, 0, "Defines the maximum angular velocity", 0.5, 0.0, 2.0)
gen.add("arrival_fading", double_t, 0, "Distance to goal position where the robot starts to fade down the linear velocity", 0.5, 0.0, 5.0)
gen.add("ang_vel_factor", double_t, 0, "Factor for angular velocity", 1.0, 0.1, 10.0)
gen.add("lin_vel_factor", double_t, 0, "Factor for linear velocity", 1.0, 0.1, 10.0)
gen.add("max_angle", double_t, 0, "The maximum angle for the linear velocity function", 20.0, 1.0, 180.0)
gen.add("max_search_radius", double_t, 0, "The maximum radius in which to search for a consecutive neighbour face", 0.4, 0.01, 2.0)
gen.add("max_search_distance", double_t, 0, "The maximum distance from the surface which is accepted for projection", 0.4, 0.01, 2.0)

exit(gen.generate("mesh_controller", "mesh_controller", "MeshController"))
